### ISSUE TITLE
Documentation of combining operators (better examples)

### ISSUE THREAD

**Issue by agowa** [mikefarah/yq#1767]
**Please describe your feature request.**
Currently, the documentation doesn't provide good examples for usage of the operators together with other operators.

For example, how to create a union after other operations again? Or how to "just sort" the output after transforming and/or flattening some parts.

**Describe the solution you'd like**
If we have data1.yml like:
(please keep to around 10 lines )

```yaml
---
pull:
  - "foo":
    - "bar"
  - "foo2":
    - "bar2"
```

And we run a command:

```bash
cat data1.yml | yq '(.pull|to_entries[].value|keys)|to_entries[].value'
```

it could output

```yaml
"foo"
"foo2"
```

The current documentation does not answer how one would extend this with another operator, like either e.g. sort or unique. So one may then look for how to create a union (as it is not a yaml list currently) and the documentation doesn't even provide an operator for that, just a command line switch. So doing union at this stage wouldn't be possible?

Expected syntax:

```bash
cat data1.yml | yq '[(.pull|to_entries[].value|keys)|to_entries[].value]|sort'
```

but this isn't intuitively discoverable within the documentation as of right now.

**Describe alternatives you've considered**
trial and error syntax, or look into the code.

**Additional context**
Even though this simple "combination" syntax may be obvious to advanced users, it is not for beginners and presents a significant challenge that could be simplified by providing better examples in the documentation for how to combine different operators (also the first thing I tried was using round parentheses, as that's what most other things use to indicate "the next operation on all of these instead of single element please")

**Comment by mikefarah** [comment]
Yeah I agree - I've been meaning to work on a 'recipes' section in the help docs. Will get to it 👍🏼

**Comment by mikefarah** [comment]
First cut: https://mikefarah.gitbook.io/yq/recipes

**Comment by agowa** [comment]
Thank you very much for this. It really clears some things up for me. Also I just learned about some things that I thought not to be possible with yq allone.

(And I learned about a very useful operator that is also a very easy pitfall `|=`. I already know that I'll always keep confusing it with the same operator from programming languages where something like `a |= b` is short for `a = a| b` (set a to a XOR b). 😅)
